From b3a2eb7b6eebf439f3f763dcc478623a6eda6051 Mon Sep 17 00:00:00 2001
From: Christophe Priouzeau <christophe.priouzeau@linaro.org>
Date: Thu, 7 Apr 2016 10:13:39 +0200
Subject: [PATCH 04/24] STM: codecparser: h265: correct frame height for
 interlaced streams

- Decoded frame height should be twice that of field height.

Change-Id: Ibbe5be661edae408d32fb5b6effe8e918e846812
Signed-off-by: Hugues FRUCHET <hugues.fruchet@st.com>
---
 gst-libs/gst/codecparsers/gsth265parser.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/gst-libs/gst/codecparsers/gsth265parser.c b/gst-libs/gst/codecparsers/gsth265parser.c
index f5ceb61..389c3ca 100644
--- a/gst-libs/gst/codecparsers/gsth265parser.c
+++ b/gst-libs/gst/codecparsers/gsth265parser.c
@@ -1697,6 +1697,9 @@ gst_h265_parse_sps (GstH265Parser * parser, GstH265NalUnit * nalu,
   /* Calculate  width and height */
   sps->width = sps->pic_width_in_luma_samples;
   sps->height = sps->pic_height_in_luma_samples;
+  if (vui && vui->field_seq_flag)
+    sps->height *=2 ;
+
   if (sps->width < 0 || sps->height < 0) {
     GST_WARNING ("invalid width/height in SPS");
     goto error;
-- 
2.7.4

